examples/gpg/gpg_keys.sh

```diff
@@ -30,6 +30,7 @@ Key-Type: RSA
 Key-Length: 2048
 Subkey-Type: RSA
 Subkey-Length: 2048
+Preferences: SHA256 SHA384 SHA512 SHA224 AES256 AES192 AES CAST5 ZLIB BZIP2 ZIP Uncompressed
```

Comment:
The whole set/show pref interface in gpg is just horrible.

I just took the values for showpref as generated from a more modern version of gpg (2.1), and CAST5 was there. No particular reason, it's listed as the last preference.

The order is important yes, the default is the first algorithm listed for each group (yuk). IOW in this case SHA256 will be used for hashsums, AES256 as the cypher, and ZLIB to compress by the generated key.

I've gone with SHA256 as the default as some older software might not support SHA512. Same as the key length, RSA4096 would be better but 2048 has more chances of working on ancient stuff.

The order of the rest won't matter much and there's no reason for listing the trucated ones in that particular order, so I can change it, how would you prefer it? 256 224 512 384 or 256 512 224 384?